chef/cookbooks/cinder/recipes/ceph.rb

```diff
@@ -17,51 +17,83 @@
 # Recipe:: ceph
 #
 
-glance_servers = search(:node, "roles:glance-server")
-if glance_servers.length > 0
-  glance_pool = glance_servers[0][:glance][:rbd][:store_pool]
-else   
-  glance_pool = nil
-end
+ceph_clients = {}
+ceph_keyrings = {}
 
 node[:cinder][:volumes].each_with_index do |volume, volid|
-  if volume['backend_driver'] == "rbd"
+  unless volume[:backend_driver] == "rbd"
+    next
+  else
+
+    ceph_conf = volume[:rbd][:config_file]
+    admin_keyring = volume[:rbd][:admin_keyring]
+    if File.exists?(admin_keyring)
+      Chef::Log.info("Using external ceph cluster for cinder #{volume[:backend_name]} backend, with automatic setup.")
+    else
+      Chef::Log.info("Using external ceph cluster for cinder #{volume[:backend_name]} backend, with no automatic setup.")
+      next
+    end
+
+    cmd = ["ceph", "-k", admin_keyring, "-c", ceph_conf, "-s"]
+    check_ceph = Mixlib::ShellOut.new(cmd)
```

Comment:
Playing with the patch: this fails the very first time we apply the barclamp because ceph-common is not installed yet.